jax/interpreters/xla.py

```diff
@@ -601,27 +601,50 @@ def _xla_callable_device(nreps, backend, device, arg_devices):
     else:
       assert False  # Unreachable given the error check in _xla_callable
 
-def _xla_callable_args(c, avals, tuple_args, replicated=None):
+# Used within _xla_callable_args and _xla_param to distinguish between None (no
+# sharding annotation set) and replicated.
+_replicated_param = object()
+
+def _xla_callable_args(
+    c, avals, tuple_args, replicated=None,
+    partitions: Optional[Sequence[Optional[Tuple[int]]]] = None):
```

Comment:
Is the inner `Tuple` a `Sequence`?

Reply:
In practice it's always a Tuple, but Sequence works. I changed it.